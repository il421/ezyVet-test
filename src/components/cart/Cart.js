import React, { Component } from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';

import Item from './Item.js';
import Checkout from './Checkout.js';
import Totals from './Totals.js';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { removeItem, setItemQty, removeAllItems } from '../../actions/cart.js';

import '../../styles/common/button.css';
import '../../styles/cart/cart.css';
import '../../styles/cart/modal.css';

export class Cart extends Component {

  state = {
    displayCheckoutModal: false
  }

  handleDisplayCheckoutModal = (active) => {
    this.setState(() => ({
      displayCheckoutModal: active
    }));
  }

  render() {
    return (
      <section className="cart">
        <div className="cart__wrapper">
          <div className="cart__actions actions">
            <button
              className="actions__checkout button"
              onClick={ () => this.handleDisplayCheckoutModal(true) }
            >
              Checkout
            </button>
            <button
              className="actions__trash button"
              onClick={ this.props.removeAllItems }
            >
              <FontAwesomeIcon icon="trash-alt" size="1x" />
            </button>
          </div>

          <div>
            {
              this.props.cart.map((item, index) => (
                <Item
                  {...item}
                  key={ index }
                  removeItem={ this.props.removeItem }
                  setItemQty={ this.props.setItemQty }
                />
              ))
            }
          </div>
        </div>

        <Totals cart={ this.props.cart }/>

        <Modal
          isOpen={ this.state.displayCheckoutModal }
          onRequestClose={ () => this.handleDisplayCheckoutModal(false) }
          closeTimeoutMS={ 200 }
          className="modal"
          ariaHideApp={ false }
        >
          <Checkout handleDisplayCheckoutModal={ () => this.handleDisplayCheckoutModal(false) } />
        </Modal>
      </section>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    cart: state.cart
  };
};

const mapDispatchToProps = (dispatch) => (
  {
    removeItem: (id) => dispatch(removeItem(id)),
    setItemQty: ({id, qty}) => dispatch(setItemQty({id, qty})),
    removeAllItems: () => dispatch(removeAllItems())
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(Cart);
