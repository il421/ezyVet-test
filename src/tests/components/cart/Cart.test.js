import React from 'react';
import { shallow } from 'enzyme';
import { Cart } from '../../../components/cart/Cart.js';
import { cart } from '../../fixtures/data';

let wrapper, removeItem, setItemQty, removeAllItems;

beforeEach(() => {
  removeItem = jest.fn();
  setItemQty = jest.fn();
  removeAllItems = jest.fn();
  wrapper = shallow(
    <Cart
      cart={ cart }
      removeItem={ removeItem }
      setItemQty={ setItemQty }
      removeAllItems={removeAllItems}
    />
  );
});

test('should render Cart without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call removeAllItems if click clear all btn', () => {
  wrapper.find('.actions__trash').simulate('click');
  expect(removeAllItems).toHaveBeenCalled();
});

