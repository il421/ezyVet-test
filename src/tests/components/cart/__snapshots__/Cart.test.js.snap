// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Cart without crashing 1`] = `
<section
  className="cart"
>
  <div
    className="cart__wrapper"
  >
    <div
      className="cart__actions actions"
    >
      <button
        className="actions__checkout button"
        onClick={[Function]}
      >
        Checkout
      </button>
      <button
        className="actions__trash button"
        onClick={[MockFunction]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="trash-alt"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="1x"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
    <div>
      <Item
        id={1}
        key="0"
        name="Sledgehammer"
        price={125.75}
        qty={1}
        removeItem={[MockFunction]}
        setItemQty={[MockFunction]}
      />
      <Item
        id={2}
        key="1"
        name="Axe"
        price={190.5}
        qty={1}
        removeItem={[MockFunction]}
        setItemQty={[MockFunction]}
      />
    </div>
  </div>
  <Totals
    cart={
      Array [
        Object {
          "id": 1,
          "name": "Sledgehammer",
          "price": 125.75,
          "qty": 1,
        },
        Object {
          "id": 2,
          "name": "Axe",
          "price": 190.5,
          "qty": 1,
        },
      ]
    }
  />
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={200}
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <Checkout
      handleDisplayCheckoutModal={[Function]}
    />
  </Modal>
</section>
`;
